webknossos/webknossos/dataset/mag_view.py

```diff
@@ -32,11 +33,15 @@
 from .view import View
 
 
-def _find_mag_path_on_disk(
-    dataset_path: Path, layer_name: str, mag_name: str, path: Optional[str] = None
+def _find_mag_path(
+    dataset_path: Path,
+    layer_name: str,
+    mag_name: str,
+    path: Optional[str | Path] = None,
 ) -> Path:
+    path = UPath(path) if path else None
     if path is not None:
-        return dataset_path / path
+        return path
 
```

Comment:
This is one of the dangerous parts of this pr imo:

I changed the functionality so that in case a mag_path is passed to the method, the path is used and not post appended to the dataset path. This was necessary to make it work with foreign mags as they have absolute paths, which would break in the `return dataset_path / path` code line.

=> the dangerous code parts are where the fourth parameter (path) is actually passed to this method. I marked the code lines